analyses/cell-type-consensus/exploratory-notebooks/03-osteosarcoma-consensus-celltypes.Rmd

```diff
@@ -0,0 +1,424 @@
+
+```{r}
+stacked_barchart(immune_results_df, fill_color = "immune_category", facet_variable = "project_id", colors = three_color_scheme)
+```
+
+When looking at all samples together we do see variation in immune cells classified and it appears that `SCPCP000017` and `SCPCP000018` have more cells classified in general and have more cells classified as immune. 
+It does appear that libraries that have more immune cell composition also have a higher percentage of non-immune cell types which could very well be a technical artifact and related to sample prep. 
+
+Just for visualization purposes, let's classify these as "hot" and "cold", where "hot" tumors have an immune composition > 5%.
+
+```{r}
+immune_results_df <- immune_results_df |> 
+  dplyr::mutate(
+    # classify hot/cold based on immune percent > 5 %
+    hot_cold = dplyr::if_else(percent_immune > .05, "hot", "cold")
+  )
+```
+
+```{r}
+stacked_barchart(immune_results_df, fill_color = "immune_category", facet_variable = "hot_cold", colors = three_color_scheme)
+```
+
+Our threshold is pretty arbitrary, but I think there are definitely some "cold" tumors where we see little to no immune involvement. 
+
+Below we can look at all immune cell types vs. all non-immune. 
+
+```{r}
+immune_results_df <- immune_results_df |> 
+  dplyr::mutate(
+    is_immune = dplyr::if_else(
+      immune_category == "immune" | top_celltypes == "Unknown",
+      consensus_annotation,
+      "non-immune"
+    )
+  )
+
+stacked_barchart(immune_results_df, fill_color = "is_immune", colors = all_celltype_colors)
+```
+
+Generally, if immune cells are present they tend to be macrophages. 
+There's also a lot of T cell populations, which makes sense since it has been previously noted that osteosarcoma has a high percentage of T cell infiltrate (https://doi.org/10.18632/oncotarget.19071).  
+
+## Is there any relationship between immune cell percentage and clinical metadata? 
+
+One thing I was curious about was whether or not there are any differences in the presence of immune cells based on clinical metadata such as primary vs. metastasis, initial diagnosis vs. recurrence, and tissue location (bone vs. soft tissue). 
+I also expect there might be some differences between single-cell and single-nuclei samples. 
+
+Below I look at the immune cell percentage stratified by each of the mentioned metadata. 
+This is just a preliminary look and if we really want to dig into this further we should consider looking at algorithms for comparing cell type composition in single-cell RNA-seq data, but that is outside the scope of this notebook. 
+
+```{r}
+# do some data wrangling on the metadata columns we want to look at 
+immune_results_df <- immune_results_df |> 
+  dplyr::mutate(
+    primary_or_metastasis = dplyr::if_else(!is.na(primary_or_metastasis), primary_or_metastasis, disease_timing),
+    disease_timing_mod = dplyr::if_else(disease_timing %in% c("Initial diagnosis", "Recurrence"), disease_timing, "other"),
```

Comment:
I think we want to add ordering here so the plots are in expected orders. I also am not sure why we seem end up with only "Recurrence" sample in the `primary_or_metastasis`? (Since it is only one, I think it could be dropped, maybe?)